tests/pos/t12494.scala

```diff
@@ -0,0 +1,29 @@
+
+trait Base {
+  protected[Base] def f: Int
+}
+object Base {
+  class Child extends Base {
+    protected[Base] def f: Int = 42   // ok
+    def test = f
+  }
+}
+
+/*
+object X {
+  // restriction in Scala 2 for local companions
+  // restriction in Scala 3 under -from-tasty
```

Comment:
What's going on with -from-tasty exactly here? If something can be compiled it should also work from tasty.

Reply:
asking for `companionModule` fails for local class. Perhaps you mean, conversely, if it fails from tasty, it must also fail always. Probably there is a convenient `is(Local)` flag check to add to the condition.

Reply:
> asking for companionModule fails for local class

Hmm that might be a bug since I could see that leading to different behavior in other situations. Also I tried uncommenting this code and running it with testCompiler but the pickling tests didn't fail.

Reply:
I'll try again, against the moving target of HEAD.